spec/controllers/preferences_controller_spec.rb

```diff
@@ -0,0 +1,120 @@
+require 'spec_helper'
+require 'helpers/pundit_helper'
+
+describe PreferencesController do
+  # name of the model for this RESTful resource
+  let(:factory) { :preference }
+  let(:url_params) { { course_id: 'section.course.id' } }
+  let(:url_params_factory) { { course_id: :course } }
+
+  # This should return the minimal set of values that should be in
+  # the session in order to pass any filters (e.g. authentication) defined
+  # in CoursesController. Be sure to keep this updated too.
+  let(:valid_session) { {} }
+
+  include PunditHelper
+
+  before(:each) do
+    stub_policy(PreferencePolicy)
+    CoursePolicy::Scope.any_instance.stub(:resolve) { Course.all }
+    PreferencePolicy::Scope.any_instance.stub(:resolve) { Preference.all }
+  end
+
+  describe do
+    before(:each) do
+      sign_in create(:user)
+    end
+
+    it_behaves_like 'a JSON resource controller:'
+  end
+
+  describe do
+    let(:preference) { create(:preference) }
+    let(:section) { preference.section }
+    let(:course) { section.course }
+    let(:user) { preference.user }
+
+    before(:each) do
+      sign_in user
+    end
+
+    describe 'get()' do
+      def get_preference
+        get :get, {
+            course_id: course.id,
```

Comment:
Use 2 spaces for indentation in a hash, relative to the start of the line where the left curly brace is.